data-tool/flows/batch_delete_flow.py

```diff
@@ -0,0 +1,444 @@
+
+
+@task
+def auth_delete(db_engine: Engine, identifiers: list):
+    with db_engine.connect() as conn:
+        with replica_role(conn):
+            entities_future = execute_query.submit(conn, {
+                'source': 'entities',
+                'params':{'business_identifier': identifiers},
+            })
+
+            plan = entities_future.result()
+
+            entity_ids = plan['entities']
+
+            # delete plans for history tables are different from version tables
+            # for this versioning method, the first record doesn't exist in history,
+            # so we should refer to the original table
+            # if the record is deleted in original table, it will be only in history table
+            query_plans = [
+                {
+                    'source': 'affiliations',
+                    'params': {'entity_id': entity_ids},
+                },
+                {
+                    'source': 'contact_links',
+                    'columns': ['id', 'contact_id'],
+                    'params': {'entity_id': entity_ids},
+                    'targets': ['contact_links', 'contacts'],
+                },
+                {
+                    'source': 'affiliations_history',
+                    'params': {'entity_id': entity_ids},
+                },
+                {
+                    'source': 'contact_links_history',
+                    'columns': ['id', 'contact_id'],
+                    'params': {'entity_id': entity_ids},
+                    'targets': ['contact_links_history', 'contacts_history'],
+                }
+            ]
+
+            query_futures = []
+            for plan in query_plans:
+                query_futures.append(
+                    execute_query.submit(conn, plan)
+                )
+
+            delete_futures = []
+            for future in query_futures:
+                delete_plans = future.result()
+                for table, ids in delete_plans.items():
+                    delete_futures.append(
+                        execute_delete_plan.submit(conn, table, ids)
+                    )
+            
+            # delete records in entities table
+            delete_futures.append(
+                execute_delete_plan.submit(conn, 'entities', entity_ids)
+            )
+
+            wait(delete_futures)
+            succeeded = sum(1 for f in delete_futures if f.state.is_completed())
+            failed = len(delete_futures) - succeeded
+            print(f'Auth delete complete for this round. Succeeded: {succeeded}. Failed: {failed}')
+
+
+@task
+def colin_delete(config, db_engine: Engine, identifiers: list):
+    with db_engine.connect() as conn:
+        plan = execute_query(conn, {
+            'source': 'corp_processing',
+            'params': {'corp_num': identifiers, 'environment': [config.DATA_LOAD_ENV]}
+        })
+
+        delete_by_ids(conn, 'corp_processing', plan['corp_processing'])
```

Comment:
`corp_processing` entries did not get deleted when i tried this locally.

The entries still exist after the flow runs.  I could see that the right environment and business identifiers were being picked up via the delete flow logging.

**corp_processing table after delete flow ran**
![image](https://github.com/user-attachments/assets/d212a345-3a40-4496-b46d-35e3a0424040)

**delete flow log snippet**
```
11:26:51.319 | INFO    | Flow run 'heavy-mastodon' - 🚀 Running round 0 to delete 5 busiesses...
11:26:51.362 | INFO    | Task run 'colin_delete-7db' - colin_delete - identifiers: ['BC0813248', 'BC0814225', 'BC0814580', 'BC0812630', 'BC0814579']
11:26:51.364 | INFO    | Task run 'colin_delete-7db' - colin delete environment - local_migr_test
11:26:51.406 | INFO    | Task run 'execute_query-d6e' - Finished in state Completed()
11:26:51.410 | INFO    | Task run 'execute_query-736' - Finished in state Completed()
11:26:51.632 | INFO    | Task run 'delete_by_ids-d46' - Delete 5 rows from corp_processing
11:26:51.664 | INFO    | Task run 'delete_by_ids-d46' - Finished in state Completed()
11:26:51.687 | INFO    | Task run 'execute_query-532' - Finished in state Completed()
11:26:51.688 | INFO    | Task run 'colin_delete-7db' - Finished in state Completed()
11:26:51.694 | INFO    | Task run 'execute_query-b64' - Finished in state Completed()
```